src/Model/Translatable/TranslatableMethods.php

```diff
@@ -56,6 +56,28 @@ public function addTranslation($translation)
     }
 
     /**
+     * Setter for translations collection
+     * 
+     * @param ArrayCollection|Array $translations
+     *
+     * @return $this
+     */
+    public function setTranslations($translations)
+    {
+        if (!$translations instanceof \Traversable && !is_array($translations)) {
+            throw new \InvalidArgumentException('$translations parameter must be iterable');
+        }
+
+        $this->translations = new ArrayCollection();
```

Comment:
woujdn't this break persistent collections? Doctrine would get lost.

Reply:
Indeed, I think you need to empty the current collection and replace its content.